src/kubernetes_operator/iam_mapping.py

```
@@ -0,0 +1,221 @@
+    with open(crd_file_path.resolve(), "r") as stream:
+        body = yaml.safe_load(stream)
+    extensions_api = client.ApiextensionsV1beta1Api()
+    crds = extensions_api.list_custom_resource_definition()
+    crds_name = {x["metadata"]["name"]: x["metadata"]["resource_version"] for x in crds.to_dict()["items"]}
+    crd_name = body["metadata"]["name"]
+    if crd_name not in crds_name.keys():
+        try:
+            extensions_api.create_custom_resource_definition(body)
+        except ValueError as err:
+            if err.args[0] != "Invalid value for `conditions`, must not be `None`":
+                raise err
+    else:
+        body["metadata"]["resourceVersion"] = crds_name[crd_name]
+        extensions_api.replace_custom_resource_definition(crd_name, body)
+
+
+def full_synchronize() -> None:
+    """Synchronize all aws-auth configmap mappings with existing IamIdentityMappings."""
+    # Get Kubernetes" objects
+    configmap = API.read_namespaced_config_map("aws-auth", "kube-system")
+    identity_mappings = custom_objects_API.list_cluster_custom_object(GROUP, VERSION, PLURAL)
+
+    identities = get_identity_mappings(configmap)
+    identities = identities if isinstance(identities, list) else list()
+
+    for identity_mapping in identity_mappings["items"]:
+        identities = ensure_identity(identity_mapping["spec"], identities)
+
+    asyncio.run(apply_identity_mappings(configmap, identities))
+
+
+def get_identity_mappings(configmap: V1ConfigMap) -> list:
+    """Get the identity mappings from the aws-auth configmap as a list.
+
+    :return identities: The combined user and role mapping list
+    """
+    try:
+        identities = []
+        identities.extend(yaml.safe_load(configmap.data["mapUsers"]))
+        identities.extend(yaml.safe_load(configmap.data["mapRoles"]))
+
+        return identities
+    except yaml.YAMLError as yaml_error:
+        logger.warning("Error loading configmap mappings: %s", yaml_error)
+        return []
```

I don't know, shouldn't we let this operator crash instead of catching it and printing a warning ? We could erase all mappings with this.

---

Forgot to change it. Will do.